pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -719,9 +748,16 @@ def get_cal_data_points(self):
                 cal_state = \
                     [k for k, idx in cal_states_rot_qb.items()
                      if idx == i][0]
-                self.cal_states_dict_for_rotation[qbn][cal_state] = \
-                    None if do_PCA and self.num_cal_points != 3 else \
-                        self.cal_states_dict[cal_state]
+                if do_PCA and self.num_cal_points != 3:
+                    self.cal_states_dict_for_rotation[qbn][cal_state] = None
+                else:
+                    if not len(self.cal_states_dict):
+                        self.cal_states_dict_for_rotation[qbn][cal_state] = None
+                        if self.get_param_value('TwoD', default_value=False):
+                            self.rotation_type = 'global_PCA'
+                    else:
+                        self.cal_states_dict_for_rotation[qbn][cal_state] = \
+                            self.cal_states_dict[cal_state]
 
     def cal_states_analysis(self):
```

Comment:
Method name not self-explanatory. A docstring might help.

Reply:
I now changed the name and added a docstring.